### PR TITLE
Add missing tests and clean up a little bit

### PR DESCRIPTION
**Pull request by elenatorro**

### CODE REVIEW

File: src/jobs/jobs.controller.js [hunk 1]
```diff
@@ -38,17 +40,31 @@ async function getAll() {
 /**
  * Transform an job from the database to a public object availiable in the API.
  */
-function _createPublicJob(job) {
+function _createPublicJob(job, teamId) {
```
Comment (IagoLast) [inline src/jobs/jobs.controller.js]: I disagree, offers should be unique, no matter the team which created them, same for the votes

File: src/jobs/jobs.controller.js [hunk 2]
```diff
@@ -38,17 +40,31 @@ async function getAll() {
...
   return {
     createdAt: job.createdAt,
     description: job.description,
     link: job.link,
     votes: {
-      upvotes: (job.votes && Object.values(job.votes).filter(text => text === 'upvote').length) || 0,
-      downvotes: (job.votes && Object.values(job.votes).filter(text => text === 'downvote').length) || 0,
+      upvotes: _getVotes.call(this, teamId, job.votes, 'upvote'),
```
Comment (IagoLast) [inline src/jobs/jobs.controller.js]: there is no need of `.call(this`

File: src/oauth/oauth.controller.js [hunk 3]
```diff
@@ -1,29 +1,33 @@
 const winston = require('winston');
 const teamsController = require('../teams/teams.controller');
+const Urls = require('../constants/urls');
 
-const OAUTH_URL = 'https://slack.com/api/oauth.access';
 const CLIENT_ID =  process.env.SLACK_CLIENT_ID;
```
Comment (IagoLast) [inline src/oauth/oauth.controller.js]: I think this should be moved to "config"

File: src/jobs/jobs.controller.js [hunk 4]
```diff
@@ -2,28 +2,30 @@ const winston = require('winston');
...
-  const existingOffer = await persistence.getOffer(offer);
-  if (!existingOffer) {
-    return persistence.saveOffer(offer);
-  }
-  return Promise.resolve(existingOffer);
+async function postJob(job) {
+  winston.info('jobs-controller:postJob', job);
+  const existingJob = await persistence.getJob(job);
+
+  return !existingJob
```
Comment (IagoLast) [inline src/jobs/jobs.controller.js]: you can change the order of the expressions in the ternary operator and remove the negation

File: src/jobs/jobs.controller.js [hunk 5]
```diff
@@ -2,28 +2,30 @@ const winston = require('winston');
...
+    ? { job: persistence.saveJob(job), existing: false }
+    : { job: existingJob, existing: true };
 }
 
 /**
- * Add a new vote to an offer.
+ * Add a new vote to a job.
  *
- * @param {object} offer
+ * @param {object} jobId
```
Comment (IagoLast) [inline src/jobs/jobs.controller.js]: is jobId an object or a string?

File: src/jobs/jobs.controller.js [hunk 6]
```diff
@@ -36,19 +38,28 @@ async function getAll() {
...
+      upvotes: _getVotes(job.votes, 'upvote'),
+      downvotes: _getVotes(job.votes, 'downvote')
     },
-    meta: job.meta
   };
 }
 
+function _getVotes(votes, voteType) {
+  return (votes &&
+    _filterByVoteType.call(this, Object.values(votes), voteType)).length || 0;
```
Comment (IagoLast) [inline src/jobs/jobs.controller.js]: no need for `.call(this`

File: src/teams/teams.controller.js [hunk 7]
```diff
@@ -1,23 +1,36 @@
...
  */
-async function save(team) {
-  winston.info('teams-controller:postTeam', team);
-  const slackTeam = persistence.saveTeam(team);
+async function save(credentials) {
+  winston.info('teams-controller:save', credentials);
 
-  return Promise.resolve(slackTeam);
+  return credentials.hasOwnProperty('team_id')
+    ? Promise.resolve(persistence.save(credentials))
```
Comment (IagoLast) [inline src/teams/teams.controller.js]: why the promise wrap?

File: src/jobs/jobs.controller.js (hunk 6 above)
Comment (elenatorro) [reply src/jobs/jobs.controller.js]: 👍

File: src/teams/teams.controller.js (hunk 7 above)
Comment (elenatorro) [reply src/teams/teams.controller.js]: ouch +2

**Comment by IagoLast** [comment]
🚀

**Pull request merged**